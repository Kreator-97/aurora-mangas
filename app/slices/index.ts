export { default as uiSlice } from './uiSlice'
export { default as authSlice } from './authSlice'
export { default as shoppingCartSlice } from './shoppingCartSlice'
