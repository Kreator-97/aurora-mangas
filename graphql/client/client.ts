import { ApolloClient, InMemoryCache, HttpLink } from '@apollo/client'
import { fetch } from 'cross-fetch'

const link = new HttpLink({
  uri: process.env.NODE_ENV === 'production' ? '/api/graphql' : 'http://localhost:3000/api/graphql',
  fetch
})

export const client = new ApolloClient({
  link,
  cache: new InMemoryCache(),
})
