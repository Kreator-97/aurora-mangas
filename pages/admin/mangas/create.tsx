import { FormEvent, useState } from 'react'
import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'

import { AppLayout } from '../../../layouts'
import { dbSeries, dbUsers } from '../../../database'
import { useForm } from '../../../hooks'
import { Serie } from '../../../interfaces'

const CLOUDINARY_DOMAIN = 'https://res.cloudinary.com'

const formValidations = {
  serie: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  imgURL: {
    validation: (value:string) => value.trim() !== '' && value.startsWith(CLOUDINARY_DOMAIN) ,
    message: 'Este campo debe de ser una URL de Cloudinary válida',
  },
  number: {
    validation: (value:string) => ( !isNaN(Number(value)) ) && value !== '',
    message: 'Este campo debe de ser un número válido',
  },
  price: {
    validation: (value:string) => ( !isNaN(Number(value)) ) && value !== '',
    message: 'Este campo debe de ser un número válido',
  },
}

interface Props {
  series: Serie[];
}

const AdminCreateSerie:NextPage<Props> = ({series}) => {
  const [showErrors, setShowErrors] = useState(false)
  const { imgURL, number, price, title, serie, formState, errors, onInputChange } = useForm({
    serie: '',
    imgURL: '',
    number : '',
    price : '',
    title : '',
  }, formValidations)


  const onSave = (e:FormEvent) => {
    e.preventDefault()
    if( Object.values(errors).some( value => value !== null) ) {
      setShowErrors(true)
      return
    }
    console.log('formulario válido')
    console.log('Hay que realizar la inserción')
    console.log({formState})
  }

  return (
    <AppLayout title="Crear manga" maxWidth='md'>
      <h1 className='title'>Agregar nuevo manga</h1>
      <form className='mx-auto mb-4 max-w-lg' onSubmit={onSave}>
        <div className='flex flex-col gap-4 mb-4'>

          <div className='flex flex-col gap-2'>
            <label htmlFor="serie" className='text-lg'>
              Seleccionar serie
            </label>
            <select
              name="serie"
              id="serie"
              className='input bg-white'
              value={ serie }
              onChange={ onInputChange }
            >
              <option
                value=""
                defaultChecked
                disabled>Selecciona un autor
              </option>
              {
                series.map((serie) => {
                  return (
                    <option key={serie.id} value={serie.id}>{serie.name}</option>
                  )
                })
              }
            </select>
            { (errors.serie && showErrors) && (<p className='text-error'>{errors.serie}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="number" className='text-lg'>
              Escribe el número de volumen
            </label>
            <input
              min={0}
              type="number"
              name="number"
              id="number"
              className='input'
              placeholder='Número'
              value={ number }
              onChange={ onInputChange }
            />
            { (errors.number && showErrors) && (<p className='text-error'>{errors.number}</p>) }
          </div>
          
          <div className='flex flex-col gap-2'>
            <label htmlFor="price" className='text-lg'>
              Escribe el precio unitario
            </label>
            <input
              min={0}
              type="number"
              name="price"
              id="price"
              className='input'
              placeholder='Escribe el precio unitario'
              value={ price }
              onChange={ onInputChange }
            />
            { (errors.price && showErrors) && (<p className='text-error'>{errors.price}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="imgURL" className='text-lg'>
              Coloca la URL de la serie (cloudinary)
            </label>
            <input
              type="text"
              name="imgURL"
              id="imgURL"
              className='input'
              placeholder='Coloca una url de cloudinary'
              value={ imgURL }
              onChange={ onInputChange }
            />
            { (errors.imgURL && showErrors) && (<p className='text-error'>{errors.imgURL}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="title" className='text-lg'>
              Escribe un titulo (opcional)
            </label>
            <input
              type="text"
              name="title"
              id="title"
              className='input'
              placeholder='Titulo opcional'
              value={ title }
              onChange={ onInputChange }
            />
          </div> 
        </div>
        <button className='btn bg-accent w-full'>Guardar</button>
      </form>
    </AppLayout>
  )
}

export default AdminCreateSerie

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)
  
  const isValid = dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const series = await dbSeries.getSeries()

  return {
    props: {
      series,
    }
  }
}
