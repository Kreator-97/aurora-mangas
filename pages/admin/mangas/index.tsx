import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'
import Link from 'next/link'

import { dbMangas, dbUsers } from '../../../database'
import { Manga } from '../../../interfaces'
import { AppLayout } from '../../../layouts'
import { Table } from '../../../components'

interface Props {
  mangas: Manga[]
}

const AdminProductPage: NextPage<Props> = ({mangas}) => {
  return (
    <AppLayout title="Series | Admin" maxWidth='lg'>
      <div className='px-2'>
        <div className='flex justify-between items-center py-4'>
          <div className='w-36'></div>
          <h1 className='title'>Mangas</h1>
          <Link href='/admin/mangas/create'>
            <button className='btn bg-accent text-sm'>Agregar manga</button>
          </Link>
        </div>
        {
          mangas.length === 0 
            ? (
              <h2 className='text-lg text-center'>No existen datos de Mangas</h2>
            )
            : (
              <Table
                columns={['Id', 'Nombre', '# Volumen', 'Precio Unitario($)', 'PublicaciÃ³n']}
              >
                {
                  mangas.map((manga) => {
                    return (
                      <tr key={manga.id}>
                        <td>{manga.id}</td>
                        <td>{manga.serie.name}</td>
                        <td>{manga.number}</td>
                        <td>{manga.price}</td>
                        <td>{manga.published}</td>
                      </tr>
                    )
                  })
                }
              </Table>
            )
        }
      </div>
    </AppLayout>
  )
}

export default AdminProductPage

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)

  const isValid = await dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const mangas = await dbMangas.getAllMangas()

  return {
    props: {
      mangas,
    }
  }
}
