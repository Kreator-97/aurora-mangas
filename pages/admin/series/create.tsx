import { FormEvent, useState } from 'react'
import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'

import { AppLayout } from '../../../layouts'
import { dbAuthors, dbUsers } from '../../../database'
import { useForm } from '../../../hooks'
import { Author } from '../../../interfaces'

const CLOUDINARY_DOMAIN = 'https://res.cloudinary.com'

const formValidations = {
  name: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  author: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  periodicy: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  finished: {
    validation: (value:string) => value.trim() !== '',
    message: 'Este campo no puede estar vacio',
  },
  imgURL: {
    validation: (value:string) => value.trim() !== '' && value.startsWith(CLOUDINARY_DOMAIN) ,
    message: 'Este campo debe de ser una URL de Cloudinary válida',
  },
}

interface Props {
  authors: Author[];
}

const AdminCreateSerie:NextPage<Props> = ({authors}) => {
  const [showErrors, setShowErrors] = useState(false)
  const { imgURL, name, sinopsis, genre, finished, author, periodicy, errors ,onInputChange, formState} = useForm({
    name: '',
    author: '',
    imgURL: '',
    sinopsis: '',
    genre: '',
    periodicy: '',
    finished: '',
  }, formValidations)

  const [ genres, setGenres ] = useState<string[]>([])

  const onSave = (e:FormEvent) => {
    e.preventDefault()
    const genre = genres.map( g => g.toUpperCase()).join(', ')

    if( !genre ) {
      setShowErrors(true)
      return
    }
    if( Object.values(errors).some( value => value !== null) ) {
      setShowErrors(true)
      return
    }
    console.log({...formState, genre: genres})
    console.log('Hay que realizar la inserción')
    console.log('formulario valido')
  }

  const addGenre = () => {
    if( !genres.includes(genre) ) {
      setGenres([...genres, genre])
    }
  }

  const removeGenre = (genre:string) => {
    setGenres(prevGenre => prevGenre.filter( g => g !== genre))
  }

  return (
    <AppLayout title="Crear serie | Admin" maxWidth='md'>
      <h1 className='title'>Agregar nueva serie</h1>
      <form className='mx-auto mb-4 max-w-lg' onSubmit={onSave}>
        <div className='flex flex-col gap-4 mb-4'>

          <div className='flex flex-col gap-2'>
            <label htmlFor="name" className='text-lg'>
              Nombre de la serie
            </label>
            <input
              type="text"
              name="name"
              id="name"
              className='input'
              placeholder='Elige el nombre de la serie'
              value={ name }
              onChange={ onInputChange }
            />
            { (errors.name && showErrors) && (<p className='text-error'>{errors.name}</p>) }
          </div>
          
          <div className='flex flex-col gap-2'>
            <label htmlFor="author" className='text-lg'>
              Seleccionar autor
            </label>
            <select
              name="author"
              id="author"
              className='input bg-white'
              value={ author }
              onChange={ onInputChange }
            >
              <option
                value=""
                defaultChecked
                disabled>Selecciona un autor
              </option>
              {
                authors.map((author) => {
                  return (
                    <option key={author.id} value={author.id}>{author.name}</option>
                  )
                })
              }
            </select>
            { (errors.author && showErrors) && (<p className='text-error'>{errors.author}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="finished" className='text-lg'>
              Estado de la serie
            </label>
            <select
              name="finished"
              id="finished"
              className='input bg-white'
              value={ finished }
              onChange={ onInputChange }
            >
              <option value="" defaultChecked disabled >Selecciona una opción</option>
              <option value="true">Finalizado</option>
              <option value="false">En progreso</option>
            </select>
            { (errors.finished && showErrors) && (<p className='text-error'>{errors.finished}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="genre" className='text-lg'>
              Añade los generos
            </label>
            <div
              className='grid gap-2'
              style={{ gridTemplateColumns: '1fr auto'}}
            >
              <select
                name="genre"
                id="genre"
                className='input bg-white'
                value={ genre }
                onChange={ onInputChange }
              >
                <option value="" defaultChecked disabled >Selecciona una opción</option>
                <option value="ACCION" >Acción</option>
                <option value="ROMANCE" >Romance</option>
                <option value="FANTASY" >Fantasia</option>
              </select>
              <button
                className='btn bg-accent'
                type='button'
                onClick={ () => addGenre() }
              >
                Agregar
              </button>
            </div>
            <div className='flex gap-2'>
              {
                genres.map((g) => {
                  return (
                    <button
                      key={g}
                      className='btn text-sm'
                      type='button'
                      onClick={() => removeGenre(g) }
                    >
                      { g } ❌
                    </button>
                  )
                })
              }
            </div>
            { (genres.length === 0 && showErrors) && (<p className='text-error'>Debes agregar un genero</p>)  }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="periodicy" className='text-lg'>
              Periodicidad
            </label>
            <select
              name="periodicy"
              id="periodicy"
              className='input bg-white'
              value={ periodicy }
              onChange={ onInputChange }
            >
              <option value="" defaultChecked disabled >Selecciona una opción</option>
              <option value="MENSUAL" >Mensual</option>
              <option value="BIMESTRUAL" >Bimestral</option>
            </select>
            { (errors.periodicy && showErrors) && (<p className='text-error'>{errors.periodicy}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="imgURL" className='text-lg'>
              Coloca la URL de la serie (cloudinary)
            </label>
            <input
              type="text"
              name="imgURL"
              id="imgURL"
              className='input'
              placeholder='Coloca una url de cloudinary'
              value={ imgURL }
              onChange={ onInputChange }
            />
            { (errors.imgURL && showErrors) && (<p className='text-error'>{errors.imgURL}</p>) }
          </div>

          <div className='flex flex-col gap-2'>
            <label htmlFor="sinopsis" className='text-lg'>
              Escribe la sinopsis (opcional)
            </label>
            <input
              type="text"
              name="sinopsis"
              id="sinopsis"
              className='input'
              placeholder='Sinopsis opcional'
              value={ sinopsis }
              onChange={ onInputChange }
            />
          </div>

        </div>
        <button className='btn bg-accent w-full'>Guardar</button>
      </form>
    </AppLayout>
  )
}

export default AdminCreateSerie

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)
  
  const isValid = dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const authors = await dbAuthors.getAuthors()

  return {
    props: {
      authors,
    }
  }
}
