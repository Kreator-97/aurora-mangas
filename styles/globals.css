@tailwind base;
@tailwind components;
@tailwind utilities;
@tailwind variants;

:root {
  box-sizing: border-box;
  --bg-color: #001E31;
  --bg-color2: #00474F;
  --stroke-color: #07695b70;
  --success: #21C728;
  --success-dark: #118A16;
  --dark: #333;
  --light: #eee;
  --header-height: 88px;
  --sm: 480px;
  --md: 720px;
  --lg: 960px;
  --xl: 1200px;
  --max: 1440px;
}

html {
  color: var(--light);
  background-image: linear-gradient( var(--bg-color), var(--bg-color2));
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
}

.main {
  min-height: calc(100vh -var(--header-height));
  margin: var(--header-height) auto 0;
}

.title {
  @apply text-xl md:text-2xl lg:text-3xl text-center py-2;
}

.btn {
  @apply px-2 py-1 font-normal text-light rounded transition-colors border border-solid border-light;
}

.btn.ghost {
  @apply text-accent bg-opacity-0 border-accent;
}

.bg-success-gradient {
  background-image: linear-gradient( var(--success), var(--success-dark))!important;
}

.btn.ghost:hover {
  @apply bg-light;
}

.active {
  @apply bg-success;
}

@media screen and (min-width: 720px) {
  .main {
    padding: 0 .5rem;
  }
}
