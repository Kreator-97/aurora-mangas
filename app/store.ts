import { configureStore } from '@reduxjs/toolkit'
import { uiSlice, authSlice, shoppingCartSlice } from './slices'

export const store = configureStore({
  reducer: {
    ui: uiSlice,
    auth: authSlice,
    cart: shoppingCartSlice
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
