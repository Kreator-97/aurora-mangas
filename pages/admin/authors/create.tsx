import { FormEvent } from 'react'
import { GetServerSideProps, NextPage } from 'next'
import { getSession } from 'next-auth/react'
import { Toaster, toast } from 'react-hot-toast'
import { useMutation } from '@apollo/client'

import { AppLayout } from '../../../layouts'
import { dbAuthors, dbUsers } from '../../../database'
import { useForm } from '../../../hooks'
import { CREATE_AUTHOR } from '../../../graphql/client/mutations'

const AdminCreateAuthor:NextPage = () => {
  const [ createAuthor ] = useMutation(CREATE_AUTHOR)
  const { author, onInputChange, onResetForm } = useForm({
    author: '',
  })

  const onSave = async (e:FormEvent) => {
    e.preventDefault()

    if( author.trim() === '' ) {
      toast.error('El campo autor es obligatorio')
      return
    }

    const result = await createAuthor({variables: {name: author}})
    onResetForm()
    toast.success(`Se ha creado el autor ${result.data.createAuthor.name}`)
  }

  return (
    <AppLayout title="Crear nuevo autor | Admin" maxWidth='md'>
      <h1 className='title'>Agregar nuevo autor</h1>
      <form
        className='mx-auto mb-4 max-w-lg'
        data-testid="form-create-serie"
        onSubmit={onSave}
      >
        <div className='flex flex-col gap-4 mb-4'>
          <div className='flex flex-col gap-2'>
            <label htmlFor="author" className='text-lg'>
              Nombre del autor
            </label>
            <input
              type="text"
              name="author"
              id="author"
              className='input'
              placeholder='Escribe el nombre del author'
              value={ author }
              onChange={ onInputChange }
            />
          </div>
        </div>

        <button type='submit' className='btn bg-accent w-full'>Guardar</button>
        <Toaster position='top-center'/>
      </form>
    </AppLayout>
  )
}

export default AdminCreateAuthor

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session = await getSession(ctx)
  
  const isValid = dbUsers.validateRole(session?.user.id || '', ['ADMIN'])
  if( !isValid ) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      }
    }
  }

  const authors = await dbAuthors.getAuthors()

  return {
    props: {
      authors,
    }
  }
}
