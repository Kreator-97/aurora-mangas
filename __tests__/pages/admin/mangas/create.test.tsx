import { expect } from '@jest/globals'
import { render } from '@testing-library/react'
import { default as AdminCreateManga } from '../../../../pages/admin/mangas/create'
import { series } from '../../../fixtures/db'
import { MockedProvider } from '@apollo/client/testing'
import { Provider } from 'react-redux'
import { store } from '../../../../app/store'

import { useSession } from 'next-auth/react'

jest.mock('next-auth/react')
jest.mock('next/router', () => {
  return {
    useRouter: () => ({query: {q: ''}})
  }
})

jest.mock('react-hot-toast', () => {
  return {
    ...jest.requireActual('react-hot-toast'),
    toast: {
      error: jest.fn(),
      success: jest.fn(),
    } 
  }
})

describe('tests on AdminCreateManga page', () => {
  (useSession as jest.Mock).mockReturnValue({
    status: 'unauthenticated',
    data: null,
  })

  test('should to match with snapshot', () => {
    const { container } = render(
      <MockedProvider mocks={[]}>
        <Provider store={store}>
          <AdminCreateManga series={series} />
        </Provider>
      </MockedProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
