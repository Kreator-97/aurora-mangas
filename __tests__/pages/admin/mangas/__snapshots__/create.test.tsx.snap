// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on AdminCreateManga page should to match with snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="content"
    >
      <svg
        color="white"
        cursor="pointer"
        data-testid="menu"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0.5"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16M4 18h16"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 150px; height: 30px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Aurora Mangas Logo"
          class="cursor-pointer"
          data-nimg="fixed"
          data-testid="app-logo"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="relative cursor-pointer"
        data-testid="cart-menu"
      >
        <svg
          color="white"
          cursor="pointer"
          data-testid="shopping-cart-menu"
          fill="currentColor"
          height="32"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="w-6 h-6 text-center font-semibold bg-accentDark bg-opacity-95 rounded-full absolute -top-2 -right-2"
        >
          0
        </span>
      </div>
      <form
        class="flex col-span-2"
        data-testid="search-form"
      >
        <input
          class="w-full rounded border-cyan-400 border p-1 text-sm text-dark"
          data-testid="search-input"
          placeholder="Encuentra tu manga favorito. Ejemplo: one punch man #1"
          type="search"
          value=""
        />
      </form>
      <svg
        color="white"
        cursor="pointer"
        data-testid="search-icon"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16l2.879-2.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242zM21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </nav>
  <div
    class="min-h-screen max-w-[360px] bg-dark fixed top-0 left-0 w-full z-50 p-4 transition-transform -translate-x-full z-[1000]"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <div
        class="w-8"
      />
      <h2
        class="text-2xl"
      >
        Bienvenido
      </h2>
      <svg
        class="cursor-pointer"
        data-testid="close-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="text-center mb-2"
    >
      Ingresa para gestionar tu cuenta
    </p>
    <div
      class="flex gap-2 mb-4"
    >
      <button
        class="btn bg-accent w-full"
      >
        Ingresar
      </button>
      <button
        class="btn ghost w-full"
      >
        Crear cuenta
      </button>
    </div>
  </div>
  <div
    class="fixed top-0 right-0 min-h-screen h-screen min-w-[360px] bg-dark z-50 transition-transform translate-x-full"
  >
    <div
      class="flex justify-between items-center border-b border-strokeLight p-4"
      style="height: 65px;"
    >
      <svg
        cursor="pointer"
        data-testid="close-shopping-cart-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <h2
        class="text-center text-2xl"
      >
        Carrito de compra
      </h2>
      <div
        class="w-7"
      />
    </div>
    <div
      class="flex flex-col gap-4 items-center overflow-y-scroll cartlist-items"
      style="padding-top: 1rem; padding-bottom: 1rem;"
    >
      <h2
        class="text-center"
      >
        No hay elementos en el carrito de compra
      </h2>
    </div>
  </div>
  <main
    class="main"
    style="max-width: var(--md);"
  >
    <h1
      class="title"
    >
      Agregar nuevo manga
    </h1>
    <form
      class="mx-auto mb-4 max-w-lg"
      data-testid="form-create-manga"
    >
      <div
        class="flex flex-col gap-4 mb-4"
      >
        <div
          class="flex flex-col gap-2"
        >
          <label
            class="text-lg"
            for="serie"
          >
            Seleccionar serie
          </label>
          <select
            class="input bg-white"
            id="serie"
            name="serie"
          >
            <option
              disabled=""
              value=""
            >
              Selecciona una serie
            </option>
            <option
              value="2baea5fb-2fd2-41ff-bf8b-f64d247a23dd"
            >
              Inuyasha
            </option>
            <option
              value="368852b4-62e7-449b-bd14-f6fd410037cc"
            >
              One Piece
            </option>
            <option
              value="543e5b82-6441-428e-b1c6-4c35db75d17b"
            >
              One Punch Man
            </option>
            <option
              value="9a4b8fa4-91cf-461f-a00c-efecfd3aae52"
            >
              Akira
            </option>
            <option
              value="9b63b0dc-adcd-40b9-8584-cdd3338555e1"
            >
              Hunter x Hunter
            </option>
          </select>
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <label
            class="text-lg"
            for="number"
          >
            Escribe el número de volumen
          </label>
          <input
            class="input"
            id="number"
            min="0"
            name="number"
            placeholder="Número"
            type="number"
            value=""
          />
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <label
            class="text-lg"
            for="price"
          >
            Escribe el precio unitario
          </label>
          <input
            class="input"
            id="price"
            min="0"
            name="price"
            placeholder="Escribe el precio unitario"
            type="number"
            value="undefined"
          />
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <label
            class="text-lg"
            for="imgURL"
          >
            Coloca la URL de la serie (cloudinary)
          </label>
          <input
            class="input"
            id="imgURL"
            name="imgURL"
            placeholder="Coloca una url de cloudinary"
            type="text"
            value=""
          />
        </div>
        <div
          class="flex flex-col gap-2"
        >
          <label
            class="text-lg"
            for="title"
          >
            Escribe un título
          </label>
          <input
            class="input"
            id="title"
            name="title"
            placeholder="Escribe un título"
            type="text"
            value=""
          />
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="flex flex-col"
          >
            <label
              class="text-lg"
              for="month"
            >
              Mes de publicación
            </label>
            <input
              class="input"
              id="month"
              max="12"
              min="1"
              name="month"
              placeholder="Introduce el mes de publicación"
              type="number"
              value=""
            />
          </div>
          <div
            class="flex flex-col"
          >
            <label
              class="text-lg"
              for="year"
            >
              Año de publicación
            </label>
            <input
              class="input"
              id="year"
              max="2030"
              min="2020"
              name="year"
              placeholder="Introduce el año de publicación"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="grid grid-cols-2 gap-2"
        >
          <div
            class="flex flex-col"
          >
            <label
              class="text-lg"
              for="stock"
            >
              Stock actual
            </label>
            <input
              class="input"
              id="stock"
              min="0"
              name="stock"
              readonly=""
              type="number"
              value="0"
            />
          </div>
          <div
            class="flex flex-col"
          >
            <label
              class="text-lg"
              for="incrementStock"
            >
              Incrementar Stock
            </label>
            <input
              class="input"
              id="incrementStock"
              min="0"
              name="incrementStock"
              placeholder="Añade el incremento de stock "
              type="number"
              value="0"
            />
          </div>
        </div>
      </div>
      <button
        class="btn bg-accent w-full"
      >
        Guardar
      </button>
      <div
        style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
      />
    </form>
  </main>
  <footer
    class="min-h-[150px] border-t border-solid border-stroke"
  >
    <h2
      class="text-xl md:text-3xl text-center p-4 mt-8"
    >
      Síguenos en nuestras redes sociales
    </h2>
    <div
      class="flex justify-center gap-8"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="facebook"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 36px; height: 36px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="twitter"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </footer>
</div>
`;
